python/lib/sift_py/grpc/transport.py

```diff
@@ -126,26 +138,36 @@ def _compute_channel_options(opts: Optional[SiftChannelConfig] = None) -> List[T
     return options
 
 
-def _metadata_interceptor(config: SiftChannelConfig) -> ClientInterceptor:
+def _metadata_interceptor(
+    config: SiftChannelConfig, metadata: Optional[Dict[str, Any]] = None
+) -> ClientInterceptor:
     """
     Any new metadata goes here.
     """
     apikey = config["apikey"]
-    metadata: Metadata = [
-        ("authorization", f"Bearer {apikey}"),
-    ]
-    return MetadataInterceptor(metadata)
+    md: Metadata = [("authorization", f"Bearer {apikey}")]
+
+    if metadata:
+        kwargs = [(key, val) for key, val in metadata.items()]
```

Comment:
Could this be a regular `for key, val in metadata.items()` and append straight to `md` rather than a list comprehension? Just to avoid creating an unnecessary list :)